src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -0,0 +1,39 @@
+package com.hibob.academy.filters
+
+import com.hibob.academy.service.SessionService.Companion.SECRET_KEY
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jws
+import io.jsonwebtoken.Jwts
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerRequestFilter
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+import org.springframework.stereotype.Component
+
+@Component
+@Provider
+class AuthenticationFilter: ContainerRequestFilter {
+
+    override fun filter(requestContext: ContainerRequestContext) {
+
+        if(requestContext.uriInfo.path == "session/login") return
+
+        val cookies = requestContext.cookies
+        val jwtCookie = cookies["JWT"]?.value
```

Comment:
please make "session/login" into a const.

the cookie name should also be a const and reused (the path name can't be reused...)